### PR TITLE
Fix follow-up form handling and URL routing issues

### PR DESCRIPTION
**Pull request by holdenk**
## Summary
This PR fixes several issues with the follow-up form submission and URL routing, including a URL pattern typo that caused crashes, missing field persistence, and form validation improvements.

## Key Changes

- **Fixed URL routing typo**: Corrected the URL pattern parameter name from `followup_semi_sekret` to `follow_up_semi_sekret` to match the view's expected kwarg, fixing crashes when accessing follow-up links with trailing slashes
- **Fixed field persistence**: Added missing `user_comments` and `appeal_result` fields to the `FollowUp` model creation in `store_follow_up_result()`, ensuring user input is properly saved
- **Improved form validation**: Added comprehensive tests for `FollowUpForm` including optional field handling, invalid choice rejection, and full payload validation
- **Enhanced document handling**: Added filtering to skip `None` entries from `MultipleFileField` uploads, preventing empty `FollowUpDocuments` rows from being created
- **Made helper parameters optional**: Updated `store_follow_up_result()` method signature to have sensible defaults (`user_comments=""`, `appeal_result=""`, `follow_up_again=False`)
- **Removed redundant code**: Eliminated unnecessary variable assignment and duplicate `save()` call in document creation

## Testing
- Added comprehensive end-to-end tests in `test_followup_view.py` covering URL routing variants, form submission, field persistence, and scheduling
- Added unit tests for `FollowUpHelper` class
- Added Selenium tests for trailing slash handling and field persistence
- Extended form validation tests with edge cases and full payload scenarios

## Notable Implementation Details
- The URL pattern now correctly handles email client quirks (trailing periods/slashes) without 404s or crashes
- The `appeal_result` field is now properly stored on both the `Denial` and `FollowUp` models
- Document upload filtering prevents database pollution from empty form submissions

https://claude.ai/code/session_014kALNVpYSTuYPh6wwr9wiF

## Summary by CodeRabbit

* **Tests**
  * Added end-to-end and integration tests covering follow-up form submission, validation, trailing-slash URL handling, persistence of comments/appeal selection, document uploads, and scheduling behavior.

* **Bug Fixes**
  * Follow-up submissions now reliably persist user comments and appeal selection; blank appeal selection is treated as pending.
  * Empty/blank uploaded documents are ignored so no empty entries are stored.
  * Scheduling now respects the “follow up again” choice.

* **Chores**
  * Made optional follow-up parameters default to sensible values.

[Review Change Stack](https://app.coderabbit.ai/change-stack/fighthealthinsurance/fighthealthinsurance/pull/794)

### CODE REVIEW

File: fighthealthinsurance/common_view_logic.py [hunk 1]
```diff
@@ -633,10 +633,11 @@ def store_follow_up_result(
...
         for document in followup_documents:
-            fd = FollowUpDocuments.objects.create(
+            if not document:
+                continue
+            FollowUpDocuments.objects.create(
                 follow_up_document_enc=document, denial=denial, follow_up_id=follow_up
             )
-            fd.save()
         denial.appeal_result = appeal_result
         denial.save()
```
[the remark this answers is not in the document]
Comment (holdenk) [reply fighthealthinsurance/common_view_logic.py]: Good catch — `ucr_refresh_actor` does indeed filter with `appeal_result__isnull=True`. Fixed in 82d3330: `store_follow_up_result` now writes `appeal_result or None`, and added `test_blank_appeal_result_leaves_denial_pending` to lock in the invariant.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: tests/sync/test_followup_view.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+        )
+        try:
+            response = self.client.get(bad_path)
+        except Exception:
+            response = None
+        # Either a 500 error page is rendered or an exception propagated;
+        # in both cases, no follow-up row is created.
+        self.assertEqual(FollowUp.objects.filter(denial_id=self.denial).count(), 0)
+        if response is not None:
+            self.assertIn(response.status_code, (200, 404, 500))
```
[the remark this answers is not in the document]
Comment (holdenk) [reply tests/sync/test_followup_view.py]: Agreed — tightened in 82d3330. The single overly-permissive test is now split into two explicit cases (`test_invalid_secret_does_not_render_form` and `test_invalid_secret_post_does_not_persist`), each setting `client.raise_request_exception = False` and asserting status 500, that the follow-up form is **not** rendered, and that no FollowUp row is written.

---
_Generated by [Claude Code](https://claude.ai/code)_
